Weight assignment fix

akilnarayan opened the pull request:
Fixes an issue where weights are not assigned if user manually sets samples.

jessdtate reviewed (commented):
this fix allows build_pce_normal.py to run, but this is the output:
<img width="637" alt="test_image" src="https://user-images.githubusercontent.com/3484631/154439202-83596183-bc5e-45b4-af1a-3e38c5a19c66.png">
this seems incorrect to me

akilnarayan commented:
That actually is correct -- but that demo uses a terribly high order (not sure why I set it that way), so I updated that, and also changed the demo to line up with the new, simplified `build_pce.py` demo syntax.
